src/main/java/io/pakland/mdas/githubstats/infrastructure/postgres/controller/PostgresMiddleware.java

```diff
@@ -0,0 +1,48 @@
+package io.pakland.mdas.githubstats.infrastructure.postgres.controller;
+
+import io.pakland.mdas.githubstats.application.MetricExporter;
+import io.pakland.mdas.githubstats.application.internal.CheckMetricsFromUserByRange;
+import io.pakland.mdas.githubstats.application.internal.GetMetricsFromUserByRange;
+import io.pakland.mdas.githubstats.domain.Metric;
+import io.pakland.mdas.githubstats.infrastructure.controller.Middleware;
+import io.pakland.mdas.githubstats.infrastructure.shell.model.ShellRequest;
+import org.springframework.stereotype.Controller;
+
+import java.io.IOException;
+import java.util.List;
+
+@Controller
+public class PostgresMiddleware extends Middleware {
+
+
+    private final CheckMetricsFromUserByRange checkMetricsFromUserByRange;
+
+    private final GetMetricsFromUserByRange getMetricsFromUser;
+
+    private final MetricExporter metricExporter;
+
+    protected PostgresMiddleware(CheckMetricsFromUserByRange checkMetricsFromUserByRange,
+                                 GetMetricsFromUserByRange getMetricsFromUser,
+                                 MetricExporter metricExporter) {
+        this.checkMetricsFromUserByRange = checkMetricsFromUserByRange;
+        this.getMetricsFromUser = getMetricsFromUser;
+        this.metricExporter = metricExporter;
+    }
+
+    @Override
+    public void execute(ShellRequest request) {
```

Comment:
We should do as we do in the `GitHubMiddleware`. Instead of passing a `ShellRequest` to the controller. What I am also realising, is that we are setting al the logic to the middleware, instead of the controller (as we do in the `GitHubMiddleware`). For consistence sake, we should choose one approach or the other.